src/mtype.h

```diff
@@ -268,8 +271,15 @@ class Type : public RootObject
     virtual bool isscope();
     virtual bool isString();
     virtual bool isAssignable();
+
+    virtual TypeStruct* isTypeStruct() { return NULL; }
+    virtual TypeClass* isTypeClass() { return NULL; }
+    virtual bool isTypeAggregate() { return this->isTypeStruct() || this->isTypeClass(); }
+    virtual TypeDArray* isTypeDArray() { return NULL; }
```

Comment:
`Dsymbol` and its derived classes don't have a field similar to `Type::ty` or `Expression::op`. So it's impossible.

When you think we should prefer `isXXX()` style in all places, I think you would need to propose removing `Type::ty` and `Expression::op`.

But I doubt that it will increase dmd code quality. At least `type->ty == Txxx` works enough.